crates/rattler_conda_types/src/channel/mod.rs

```diff
@@ -198,6 +198,10 @@ impl Channel {
                 }
             }
         } else {
+            // Validate that the channel is a valid name
+            if channel.contains(|c| c == ':' || c == '\\') {
```

Comment:
```suggestion
            if channel.contains(&[':', '\\']) {
```

Reply:
Maybe we can include a list of valid characters? `A-Za-z0-9-_`?

Reply:
Something like this?: 
```rs
if !channel.chars().all(|c| c.is_alphanumeric() || c == '-' || c == '_') {
    return Err(ParseChannelError::InvalidName(channel.to_owned()));
}
```
Without the Lenient option I would be afraid of breaking some working systems. The previous change was needed to avoid breaking further parsing. 
What do you think?

Reply:
Added your suggestion in https://github.com/conda-incubator/rattler/pull/792/commits/7f653af74bbabcfb40da0fa1977dc323e264dab